epsagon/aws_v2_wrapper.go

```diff
@@ -17,12 +18,13 @@ func WrapAwsV2Service(svcClient awsFactories.AWSClient, args ...context.Context)
 	apiOptions := extractAPIOptions(svcClient)
 	awsCall := &awsFactories.AWSCall{}
 	currentTracer := ExtractTracer(args)
-
-	awsFactories.AddMiddlewareFuncs(
-		apiOptions,
-		awsFactories.InitializeMiddleware(awsCall, currentTracer, completeEventData),
-		awsFactories.FinalizeMiddleware(awsCall, currentTracer),
-	)
+	if currentTracer != nil {
```

Comment:
When might currentTracer be nil? Is that a valid use case, where the user still see traces later on?
If not, I think we should fail fast with a message as informative as possible - to let the user know that they configured something wrong and will not see traces until they resolve it.

Reply:
it can be on any case of not using Epsagon correctly, or on Epsagon internal failures. He won't be able to see the AWS calls, but might (depends on the case) see a trace.
I disagree about the fail fast - we shouldn't make the user code fail, as it can happen also on our internal failures (we handle it the same way in other instrumentations). In general, in any case we shouldn't make the user code fail